runtime/src/gpu/common/rocm-utils.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020-2023 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.  *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __HIP_PLATFORM_AMD__
+#define __HIP_PLATFORM_AMD__
+#endif
+#include <hip/hip_common.h>
+#include <hip/hip_runtime.h>
+
+#if __has_include(<rocm-core/rocm_version.h>)  // 5.x wants this
```

Comment:
Oh geez, so unfortunate we have to jump through these hoops. TIL `__has_include`. It looks like this is C++17 <https://en.cppreference.com/w/cpp/preprocessor/include>, we're still on C++14 right?

Reply:
17 vs 14 is more of a compiler thing AFAIK. This is runtime code that's compiled with a different compiler (HOST vs TARGET). Based on https://stackoverflow.com/questions/142877/can-the-c-preprocessor-be-used-to-tell-if-a-file-exists (for better or worse) most compilers have already supported this for a while. In my tests across 2 machines with 4 different ROCms I haven't seen this causing any issues.